extensions/common/context.cc

```diff
@@ -295,13 +277,75 @@ flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer() {
   return fbb.Release();
 }
 
+PeerNodeInfo::PeerNodeInfo(const std::string_view peer_metadata_id_key,
+                           const std::string_view peer_metadata_key) {
+  found_ = getValue({peer_metadata_id_key}, &peer_id_);
+  if (found_) {
+    getValue({peer_metadata_key}, &peer_node_);
```

Comment:
I think we may want to send just the ID and not node once Metadata Discovery is there. It would be used as a lookup key.

Reply:
Yeah, that sounds right to me.

Reply:
Although that relies on metadata service would have all peer information, I am not sure that would be the case, for example multi cluster with multiple control planes.